designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputIconCard.kt

```diff
@@ -0,0 +1,350 @@
+) {
+    InputShell(
+        modifier = modifier.testTag("ICON_CARDS_INPUT_$testTag"),
+        title = title,
+        state = state,
+        primaryButton = null,
+        secondaryButton = null,
+        isRequiredField = isRequired,
+        inputField = {
+            VerticalGrid(
+                columns = itemCount,
+                itemCount = data.size,
+                horizontalArrangement = Arrangement.spacedBy(Spacing.Spacing8),
+                verticalArrangement = Arrangement.spacedBy(Spacing.Spacing8),
+                modifier = Modifier.padding(
+                    PaddingValues(
+                        start = Spacing.Spacing0,
+                        top = Spacing.Spacing8,
+                        end = Spacing.Spacing12,
+                        bottom = Spacing.Spacing8,
+                    ),
+                ).testTag("ICON_CARD_INPUT_MATRIX_$testTag"),
+            ) { index ->
+                val iconCardData = data[index]
+                MatrixIconCard(
+                    data = iconCardData,
+                    onClick = {
+                        onSelectionChanged(iconCardData)
+                    },
+                    selected = iconCardData == selectedData,
+                    enabled = state != InputShellState.DISABLED,
+                    modifier = Modifier.testTag("MATRIX_ICON_CARD_$testTag"),
+                )
+            }
+        },
+        supportingText = {
+            supportingText?.forEach { label ->
+                SupportingText(
+                    label.text,
+                    label.state,
+                    modifier = Modifier.testTag("ICON_CARDS_INPUT_" + testTag + "_SUPPORTING_TEXT"),
+                )
+            }
+        },
+        legend = {
+            legendData?.let {
+                Legend(legendData, Modifier.testTag("ICON_CARDS_INPUT_" + testTag + "_LEGEND"))
+            }
+        },
+    )
+}
+
+/**
+ * DHIS2 icon card sequential input component
+ * @param title: Label of the component
+ * @param data: List of [IconCardData] to show
+ * @param layoutType: [IconCardLayoutType] to render list in, Matrix or Sequential. Default is [IconCardLayoutType.Matrix]
+ * @param selectedData: Selected [IconCardData], renders selected UI around that item
+ * @param state: [InputShellState]
+ * @param supportingText: List of [SupportingTextData] that manages all the messages to be shown
+ * @param legendData: [LegendData] to be render below the input shell
+ * @param isRequired: Mark this input as marked
+ * @param onSelectionChanged: Callback to receive new selected item
+ */
+@Composable
+fun InputSequential(
+    title: String,
+    data: List<IconCardData>,
+    selectedData: IconCardData? = null,
+    modifier: Modifier = Modifier,
+    state: InputShellState = UNFOCUSED,
+    supportingText: List<SupportingTextData>? = null,
+    legendData: LegendData? = null,
+    isRequired: Boolean = false,
+    testTag: String = "",
```

Comment:
Yeah, I felt like it would be useful for testing certain elements in a list of items. But I am fine with removing them from children composables. Something I wanted to try that's all.

Reply:
I think it's a good idea 👍